crates/ruff/src/rules/pylint/rules/duplicate_values.rs

```diff
@@ -0,0 +1,45 @@
+use ruff_python_ast::comparable::ComparableExpr;
+use rustc_hash::FxHashSet;
+use rustpython_parser::ast::{self, Expr, Ranged};
+use std::hash::BuildHasherDefault;
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+
+use crate::checkers::ast::Checker;
+
+#[violation]
+pub struct DuplicateValues {
+    value: String,
+}
+
+impl Violation for DuplicateValues {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let DuplicateValues { value } = self;
+        format!("Duplicate value `{value}` in set.")
+    }
+}
+
+/// PLW0130
+/// "This message is emitted when a set contains the same value two or more times.",
+pub(crate) fn duplicate_values(checker: &mut Checker, elts: &Vec<Expr>) {
+    let mut seen_values: FxHashSet<ComparableExpr> =
+        FxHashSet::with_capacity_and_hasher(elts.len(), BuildHasherDefault::default());
+
+    for elt in elts {
+        if let Expr::Constant(ast::ExprConstant { value, .. }) = elt {
+            let comparable_value: ComparableExpr = elt.into();
+
+            if seen_values.contains(&comparable_value) {
```

Comment:
Nit: `seen_values.insert` will return a boolean that indicates whether the item already exists in the `HashSet`. So you can both check for duplicates and update the set in a single operation.